<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />

<meta name="description" content="The Shen Lab - Laboratory of Yufeng Shen, Ph.D. - Center for Computational Biology and Bioinformatics, Columbia Initiative in Systems Biology, and Department of Biomedical Informatics, Columbia University - Research focus: DNA sequencing, genome assembly computational genomics, statistical genetics, pharmacogenomics" />
<meta name="keywords" content="Shen, Columbia University, Computational Genomics, Bioinformatics, genomics, genome, sequencing, pharmacogenomics" />

<title>Shen Lab | Research</title>
<link rel="stylesheet" type="text/css" href="web.css">
</head>

<body>

<div id="container">
  <div id="header">    
   
        
      <h1>Shen Lab @ Columbia</h1>



    <div id="menu">
      <ul>
       <li><a href="index.html">Home</a></li>
      <li><a href="people.html">People</a></li>
      <li class="active"><a href="research.html">Research</a></li>
      <li><a href="papers.html">Papers</a></li>
      <li><a href="software.html">Software</a></li>
      <li><a href="contact.html">Contact</a></li>
    
      </ul>
      <div></div>
    </div>
  </div>

  <div id="mainContent">

<h3> Research </h3>



<p>	We are interested in developing and applying computational methods to study human genetics and diseases. Our work is at the interface between biology, statistics, and computer science. Here are four specific areas: 
<a href="#sequencing">Genome sequencing and assembly</a> ;  <a href="#genetics">Genetic mapping of disease genes</a> ;  <a href="#MHC"> Autoimmunity and MHC</a> ;  <a href="#Pharmacogenomics"> Pharmacogenomics</a>
</p>


<h4 id="sequencing"> Genome sequencing and assembly </h4>

<p>Genome sequencing and assembly is at the upstream of studies in genetics and systems biology. The rapid progress in high-throughput sequencing technologies fostered many novel computation problems. We are developing methods to infer genetic variants, formulating new applications of <em>de novo</em> genome assembly, and making tools that facilitate designs of sequencing studies. </p>

<h4 id="genetics"> Genetic mapping of disease genes </h4>

<p>Genetic mapping is a powerful approach to uncover genetic variants that cause disease or increase the risk of diseases. Classic genetic mapping methods, including family-based linkage for Mendelian diseases and population-based association for complex diseases, are based on genetic markers (which are observed in the experiments) that are correlated with the causal variants (which are not observed). Additionally, these two methods are best powered to find genetic variants from two extremes: very rare but high penetrant variants for Mendelian disease, and common but low penetrant variants for complex diseases.  Now using genome-wide sequencing, we can directly observe potential causal variants in the studied individuals. We can also access variants from the previously overlooked uncommon-frequency / moderate-penetrant spectrum. We are interested to solve novel statistical problems for mapping disease genes in this context. We are developing tools for both monogenetic and polygenetic diseases that integrate statistical evidences with functional information, such as conservation, expression and methylation. </p>

<p>There are evidences that complex diseases are associated with many common variants with very small effects, which cannot be addressed by conventional genetic mapping approaches without astronomical number of subjects. Yet finding these variants is still valuable for advancing our understandings of biological mechanism of the diseases. We are interested in developing statistical models to aggregate small effects of individual variants through pathways and networks.  </p>

<h4 id="MHC">Autoimmunity and MHC </h4>

<p>The immune response is one of the most fascinating processes in human body. The major histocompatibility complex (MHC) region is the key player in the immune system, and has been implicated in all major autoimmune diseases and many other diseases. But it is extremely polymorphic in human population and highly complex, mostly due to its crucial function of presenting pathological antigens. The high polymorphic rate and complex structure makes it one the most difficult regions to analyze from sequencing data. We are developing methods to fully describe this genomic region based on sequencing data, and apply them to better understand the genetics of autoimmune and related diseases. </p>

<h4 id="Pharmacogenomics"> Pharmacogenomics </h4>

<p>In collaborating with the International Serious Adverse Events Consortium (<a href="http://www.saeconsortium.org/">iSAEC</a>), we are conducting genome-wide association and sequencing studies of various idiosyncratic adverse drug reactions (ADR). The ultimate goal is to find major genetic variants that can effectively predict the risk or the lack of risk for developing ADRs, and therefore lead to personalized treatment.  We are currently pursuing studies in serious skin rush (Stevens-Johnson Syndrome) and hypersensitivity, drug-induced liver injury (DILI), and elongate QT interval or Torsades de pointes. </p>



<h3>Bookmarks</h3>

 <p> &nbsp; ➡ &nbsp; &nbsp; <a href="http://www.cs.indiana.edu/mit.research.how.to/mit.research.how.to.html">Research 101</a></p>
 <p> &nbsp; ➡ &nbsp; &nbsp; <a href="http://www.nature.com/naturejobs/2011/110707/full/nj7354-129a.html">Writing</a></p>
 
 

<h3>Colleagues and Collaborators </h3>

<ul>
<li><a href="http://www.cs.columbia.edu/~itsik/">Itsik Pe'er</a>, Columbia University, Department of Computer Science

<li><a href="http://www.liv.ac.uk/pharmacogenetics/">Munir Pirmohamed</a>, University of Liverpool, Clinical Pharmacology, Pharmacogenetics
<li><a href="http://atgu.mgh.harvard.edu/">Mark Daly</a>, Massachusetts General Hospital, The Analytic and Translational Genetics Unit; the Broad Institute 
of Harvard and MIT.


<li>Matt Nelson, GSK, Statistical Genetics
<li><a href="http://icg.cpmc.columbia.edu/faculty_Tycko.htm">Ben Tycko</a>, Columbia University Medical Center
<li><a href="http://www.ncl.ac.uk/biomedicine/research/groups/profile/ann.daly">Ann Daly</a>, Newcastle University, Pharmacogenetics
<li><a href="http://genome.wustl.edu/people/weinstock_george">George Weinstock</a>, Washington University, the Genome Institute. 
<li><a href="http://openwetware.org/wiki/Fuli's_Lab">Fuli Yu</a>, Baylor College of Medicine, the Human Genome Sequencing Center
<li><a href="http://www.hgsc.bcm.tmc.edu/content-home-HGSC_director-x.hgsc">Richard Gibbs</a>, Baylor College of Medicine, the HumanGenome Sequencing Center
<li><a href="http://eostrowski.squarespace.com/">Elizabeth Ostrowski</a>, Baylor College of Medicine
<li><a href="http://www.mailman.columbia.edu/our-faculty/profile?uni=sw2206">Shuang Wang</a>, Columbia University, Mailman School of Public Health
</ul>



      

  </div>
  
 
  <div id="footer">
      <p> </p>
  </div>
  
</div>

<script src="http://www.google-analytics.com/urchin.js" type="text/javascript">
</script>
<script type="text/javascript">
_uacct = "UA-400081-5";
urchinTracker();
</script>

</body>
</html>
