<html>
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
  <Meta http-equiv="Content-Style-Type" content="text/css">
  <title>CANOES | detecting CNVs from exome sequencing</title>
  <style type="text/css">
  body {
   font-family: "PT serif", Georgia, serif;
   font-size: 1.2em;
   padding: 60px;
   width: 800px;  
  }
  code {
   font-family: Consolas, Monaco, Courier;
   font-size: 14px;
   padding-left: 0px;
   background: #ffff99;
  }  
  li {
    margin-bottom: 1em;
  }
  </style>
</head>
<body>

<h1><i>CANOES</i></h1> 

<h3>Detecting CNVs from exome sequencing data</h3>

<h4><center> <a href="#Tutorial">Tutorial</a> &nbsp; |  &nbsp; <a href="#Contact">Contact</a>  &nbsp; |  &nbsp; <a href="#Cite">Cite</a>  </center> </h4>

<a name="Tutorial"></a><h4>Tutorial</h4>
<p>To run CANOES, you will need</p>

<ol><li>bedtools, available from <a href="http://bedtools.readthedocs.org/en/latest/">http://bedtools.readthedocs.org/en/latest/</a></li>
<li>R, with packages nnls, Hmisc, mgcv and plyr installed.  You can install these by executing the following command:  <code>install.packages(c("nnls", "Hmisc", "mgcv", "plyr"))</code>.  </li>
<li>GATK</li>
<li><a href="CANOES.R">CANOES.R</a></li>
</ol><p>CANOES requires a data frame with the coordinates, GC content and read count per sample for each exome capture region.</p>

<p>If bamlist.txt is a file with the path of a bam on each line and probes.bed is a bed file with the coordinates of the exome capture regions, bedtools can generate read counts with the following command, entered at the LINUX prompt:</p>

<pre><code>&gt; bedtools multicov -bams `cat bamlist.txt` -bed probes.bed -q 20 &gt; canoes.reads.txt
</code></pre>

<p>Make sure the contig names in probes.bed correspond to those in the header of the bam files.  The above command will save the read counts in the file canoes.reads.txt.</p>

<p>If FASTA is the path to your reference file and TEMP is a temporary directory, enter the following command at the LINUX prompt to use GATK to calculate the GC content for each exome capture region:</p>

<pre><code>&gt; java -Xmx2000m -Djava.io.tmpdir=TEMP -jar ./Sting/dist/GenomeAnalysisTK.jar -T GCContentByInterval -L probes.bed -R FASTA -o gc.txt
</code></pre>

<p>Again, the contig names in probes.bed should correspond to those in the FASTA.fai file.  </p>

<p>We can then use R to stitch together a data frame with all the required information for CANOES.  You can follow along with the two sample files <a href="gc.txt">gc.txt</a> and <a href="canoes.reads.txt">canoes.reads.txt</a>, which contain gc content and read count information for probes on chromosome 22 for 26 samples.</p>

<p>First, read in the data:</p>

<pre><code>&gt; gc &lt;- read.table("gc.txt")$V2
&gt; canoes.reads &lt;- read.table("canoes.reads.txt")
</code></pre>

<p>Now, rename the columns of canoes.reads.  Normally, one would use actual sample names instead of S1, S2, &#8230;</p>

<pre><code>&gt; sample.names &lt;- paste("S", seq(1:26), sep="")
&gt; names(canoes.reads) &lt;- c("chromosome", "start", "end", sample.names)
</code></pre>

<p>Create a vector of consecutive target ids:</p>

<pre><code>&gt; target &lt;- seq(1, nrow(canoes.reads))
</code></pre>

<p>Combine the data into one data frame:</p>

<pre><code>&gt; canoes.reads &lt;- cbind(target, gc, canoes.reads)
</code></pre>

<p>Now we can call CNVs in the samples.</p>

<pre><code># execute this command from the directory where you have saved CANOES.R
# first make sure you have the packages nnls, Hmisc, mgcv and plyr installed 
# you can install these with the command install.packages(c("nnls", "Hmisc", "mgcv", "plyr"))
&gt; source("CANOES.R")

# create a vector to hold the results for each sample
&gt; xcnv.list &lt;- vector('list', length(sample.names))

# call CNVs in each sample
&gt; for (i in 1:length(sample.names)){
    xcnv.list[[i]] &lt;- CallCNVs(sample.names[i], canoes.reads)
  }

# combine the results into one data frame
&gt;  xcnvs &lt;- do.call('rbind', xcnv.list)

# inspect the first two CNV calls
&gt; head(xcnvs, 2)
  SAMPLE CNV             INTERVAL     KB CHR   MID_BP    TARGETS NUM_TARG MLCN Q_SOME
1     S2 DEL 22:25713988-25756059 42.071  22 25735024 1132..1137        6    1     99
2     S3 DEL 22:24373138-24384231 11.093  22 24378684   936..942        7    0     44
</code></pre>

<p>Now we can plot all the CNV calls.</p>

<pre><code>&gt; pdf("CNVplots.pdf")
&gt; for (i in 1:nrow(xcnvs)){
    PlotCNV(canoes.reads, xcnvs[i, "SAMPLE"], xcnvs[i, "TARGETS"])
  }
&gt; dev.off()
</code></pre>
<p>Example output: <a href="CNVplots.pdf">CNVplots.pdf</a>. </p>

<p>Finally, we can genotype a sample to determine the evidence that the sample has a CNV at any interval.</p>

<pre><code># genotype all the CNV calls made above in sample S2
&gt; genotyping.S2 &lt;- GenotypeCNVs(xcnvs, "S2", canoes.reads)
# inspect the genotype scores for the first two CNV calls
&gt; head(genotyping.S2, 2)
                                 INTERVAL NQDel SQDel NQDup SQDup
22:25713988-25756059 22:25713988-25756059     0    99    99     0
22:24373138-24384231 22:24373138-24384231    86     0    61     0
</code></pre>

<p>The SQDel score for the first CNV is high (99) because this CNV was called in sample S2.  The second CNV was called in sample S3, so the SQDel and SQDup quality scores for sample S2 are 0.</p>
	
	<hr />
	
<a name="Contact"></a>

	<h4>Contact</h4>
 <p>Daniel Backenroth (db2175 at columbia.edu) or Yufeng Shen ( ys2411 at columbia.edu) </p>
 
 <hr />
 <a name="Cite"></a>

 <h4>How to cite:</h4>
 <p>Backenroth D, Homsy J, Murillo LR, Glessner J, Lin E, Brueckner M, Lifton R, Goldmuntz E, Chung WK, Shen Y, (2014) CANOES: Detecting rare copy number variants from whole exome sequencing data, Nucleic Acids Research,  doi: 10.1093/nar/gku345  <a href="http://www.ncbi.nlm.nih.gov/pubmed/24771342">PMID: 24771342</a> </p>
 
 <hr />
 <h4><a href="http://www.columbia.edu/~ys2411/">Shen Lab</a></h4>

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-400081-8', 'columbia.edu');
  ga('send', 'pageview');
</script>
</body>

</html>


